### PR TITLE
feat: add flatbuffer map support

### PR DESCRIPTION
**Pull request by dudantas**

### CODE REVIEW

File: source/kmap_writer.cpp [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+		0,
+		0,
+		0,
+		createActionOffSet
+	);
+
+	return Kmap::CreateItem(
+		builder,
+		0,
+		item.getID(),
```
Comment (lgrossi) [inline source/kmap_writer.cpp:199]: Shouldn't be id first then 0?

File: source/kmap_writer.cpp [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+	}
+
+	House *house = map.houses.getHouse(tile.getHouseID());
+	auto houseInfoOffset = house ? Kmap::CreateHouseInfo(
+		builder,
+		tile.getHouseID(),
+		house->getEmptyDoorID()
+	) : 0;
+
+	auto createActionOffSet = buildActionAttributes(*ground, pos);
```
Comment (lgrossi) [inline source/kmap_writer.cpp:159]: This doesn't exist anymore

File: source/kmap_writer.cpp [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+	return pos.x < areaPos->x ||
+		pos.x >= areaPos->x + 256 ||
+		pos.y < areaPos->y ||
+		pos.y >= areaPos->y + 256 ||
+		pos.z != areaPos->z;
+}
+
+FlatOffSetItem KmapWriter::buildGround(Item &item, const Position &pos)
+{
+	auto createActionOffSet = buildActionAttributes(item, pos);
```
Comment (lgrossi) [inline source/kmap_writer.cpp:185]: Nor this

File: source/kmap_writer.cpp (hunk 1 above)
Comment (dudantas) [reply source/kmap_writer.cpp:197]: image
item id is the second

Comment (lgrossi) [reply source/kmap_writer.cpp:197]: Ok, you're right my bad

**Pull request closed without merging**